network/transports_test.go

```diff
@@ -100,3 +101,41 @@ func TestDialWithBackoff(t *testing.T) {
 	}
 	c.Close()
 }
+
+func TestDialTLSWithBackoff(t *testing.T) {
+	tlsConf := &tls.Config{
+		InsecureSkipVerify: true, // FIXME: Is it possible to set false for the test server?
+		ServerName:         "example.com",
+		MinVersion:         tls.VersionTLS12,
+	}
+	// Nobody's listening on a random port. Usually.
+	c, err := DialTLSWithBackOff(context.Background(), "tcp4", "127.0.0.1:41482", tlsConf)
+	if err == nil {
+		c.Close()
+		t.Error("Unexpected success dialing")
+	}
+
+	// Make the test short.
+	bo := backOffTemplate
+	bo.Steps = 2
+
+	// Timeout. Use special testing IP address.
+	c, err = dialBackOffHelper(context.Background(), "tcp4", "198.18.0.254:8888", bo, tlsConf)
+	if err == nil {
+		c.Close()
+		t.Error("Unexpected success dialing")
+	}
+	const expectedErrPrefix = "timed out dialing"
+	if err == nil || !strings.HasPrefix(err.Error(), expectedErrPrefix) {
```

Comment:
You just checked `err` on 124.

Also, you don't check the error message for the port test -- maybe move this to a local function and use it for both connections (`verifyFailedConnection`)?